src/js/player.js

```diff
@@ -242,6 +249,126 @@ class Player extends Component {
     return el;
   }
 
+  width(value) {
+    return this.dimension('width', value);
+  }
+
+  height(value) {
+    return this.dimension('height', value);
+  }
+
+  dimension(dimension, value) {
+    let property = dimension + '_';
+
+    if (value === undefined) {
+      return this[property] || 0;
+    }
+
+    if (value === '') {
+      // If an empty string is given, reset the dimension to be automatic
+      this[property] = undefined;
+    } else {
+      this[property] = parseFloat(value);
+
+      if (isNaN(this[property])) {
+        Lib.log.error(`Improper value "${value}" supplied for for ${dimension}`);
+      }
+    }
+
+    this.updateStyleEl_();
+    return this;
+  }
+
+  fluid(bool) {
+    if (bool === undefined) {
+      return !!this.fluid_;
+    }
+
+    this.fluid_ = !!bool;
+
+    if (bool) {
+      this.addClass('vjs-fluid');
+    } else {
+      this.removeClass('vjs-fluid');
+    }
+  }
+
+  aspectRatio(ratio) {
+    if (ratio === undefined) {
+      return this.aspectRatio_;
+    }
+
+    // Check for width:height format
+    if (!/^\d+\:\d+$/.test(ratio)) {
+      throw new Error('Improper value suplied for aspect ratio. The format should be width:height, for example 16:9.');
+    }
+    this.aspectRatio_ = ratio;
+
+    // We're assuming if you set an aspect ratio you want fluid mode,
+    // because in fixed mode you could calculate width and height yourself.
+    this.fluid(true);
+
+    this.updateStyleEl_();
+  }
+
+  updateStyleEl_() {
+    let width;
+    let height;
+    let aspectRatio;
+
+    // The aspect ratio is either used directly or to calculate width and height.
+    if (this.aspectRatio_ !== undefined && this.aspectRatio_ !== 'auto') {
+      // Use any aspectRatio that's been specifically set
+      aspectRatio = this.aspectRatio_;
+    } else if (this.videoWidth()) {
+      // Otherwise try to get the aspect ratio from the video metadata
+      aspectRatio = this.videoWidth() + ':' + this.videoHeight();
+    } else {
+      // Or use a default. The video element's is 2:1, but 16:9 is more common.
+      aspectRatio = '16:9';
+    }
+
+    // Get the ratio as a decimal we can use to calculate dimensions
+    let ratioParts = aspectRatio.split(':');
+    let ratioMultiplier = ratioParts[1] / ratioParts[0];
+
+    if (this.width_ !== undefined) {
+      // Use any width that's been specifically set
+      width = this.width_;
+    } else if (this.height_ !== undefined) {
+      // Or calulate the width from the aspect ratio if a height has been set
+      width = this.height_ / ratioMultiplier;
+    } else {
+      // Or use the video's metadata, or use the video el's default of 300
+      width = this.videoWidth() || 300;
+    }
```

Comment:
If you don't add a width/height to the player, these styles don't get added, so you only have to override the default dimensions in `.video-js`. If  you do set a width/height, this sets it using a new single classname.
```
.vjs_video_292-dimensions { width: 300px; height: 168.75px; }
```
If a user's goal is to set the dimensions of the player in CSS somewhere else, they just have to do *one* of the following:
- don't set a width/height or aspect ratio on the player
- use a stronger selector than a single classname, e.g. ID

I tried to set the lowest bar I could, and this gets pretty close to how the video element works. Trying to figure out what users have set elsewhere would get really complicated and I don't think would add a ton of value.